packages/eslint-plugin/src/rules/prefer-ts-expect-error.ts

```diff
@@ -8,44 +15,58 @@ export default util.createRule<[], MessageIds>({
   meta: {
     type: 'problem',
     docs: {
-      description:
-        'Recommends using `// @ts-expect-error` over `// @ts-ignore`',
+      description: 'Recommends using `@ts-expect-error` over `@ts-ignore`',
       category: 'Best Practices',
       recommended: false,
     },
     fixable: 'code',
     messages: {
       preferExpectErrorComment:
-        'Use "// @ts-expect-error" to ensure an error is actually being suppressed.',
+        'Use "@ts-expect-error" to ensure an error is actually being suppressed.',
     },
     schema: [],
   },
   defaultOptions: [],
   create(context) {
-    const tsIgnoreRegExp = /^\/*\s*@ts-ignore/;
+    const tsIgnoreRegExp = /(^|\n)((\*)*(\/)*\s*(\/\*)*\s*)*\s*@ts-ignore/;
     const sourceCode = context.getSourceCode();
 
+    function getLastCommentLine(comment: TSESTree.Comment): string {
+      if (comment.type === AST_TOKEN_TYPES.Line) {
+        return comment.value;
+      }
+
+      // For multiline comments - we look at only the last line.
+      const commentlines = comment.value.split('/\n');
+      return commentlines[commentlines.length - 1];
+    }
     return {
       Program(): void {
         const comments = sourceCode.getAllComments();
-
         comments.forEach(comment => {
-          if (comment.type !== AST_TOKEN_TYPES.Line) {
-            return;
-          }
+          if (tsIgnoreRegExp.test(getLastCommentLine(comment))) {
+            const lineCommentRuleFixer = (fixer: RuleFixer): RuleFix =>
+              fixer.replaceText(
+                comment,
+                `//${comment.value.replace('@ts-ignore', '@ts-expect-error')}`,
+              );
+
+            const blockCommentRuleFixer = (fixer: RuleFixer): RuleFix =>
+              fixer.replaceText(
+                comment,
+                `/*${comment.value.replace(
+                  '@ts-ignore',
+                  '@ts-expect-error',
+                )}*/`,
+              );
 
-          if (tsIgnoreRegExp.test(comment.value)) {
             context.report({
               node: comment,
               messageId: 'preferExpectErrorComment',
-              fix: fixer =>
-                fixer.replaceText(
-                  comment,
-                  `//${comment.value.replace(
-                    '@ts-ignore',
-                    '@ts-expect-error',
-                  )}`,
-                ),
+              fix:
+                comment.type === AST_TOKEN_TYPES.Line
+                  ? lineCommentRuleFixer
+                  : blockCommentRuleFixer,
```

Comment:
I know the old fixer replaced the entire comment, but it might be better for us to just replace a substring instead.

`fixer.replaceTextRange` will let you do that.
You can construct the range from the range of the comment node + the index within the comment text.

This would save us having to have multiple a fixer for each case.

Reply:
I like this suggestion - I was looking for something similar couldn't find it.

Will wait for our discussion on other thread to resolve before - I change this.

Reply:
I tried this but unfortunately for comment this is not as trivial as I thought. The comment returned by `sourceCode.getAllComments` has a few characters stripped, when setting the `range` in `replaceTextRange` , we need to account for them separately.

The 2 instances where characters were getting stripped are - 
1. When trying to get the new range we needed to account for 2 additional characters either `//` or `/*` at the start.
1. With line breaks (`\n`) in multi line comments leading to again adjusting the index.

We were ending up with the same distinction in range calculation instead of fixers. 

In the end I thought having 2 fixers seemed more simpler and aligned with having 2 regex for identifying `@ts-ignore`.

If you additional thoughts around this - would be happy to implement them 👍

Reply:
Also then there were also super wired cases where an actual comment - 
```
{
  /*
        just a comment containing @ts-ignore somewhere in a block
      */
}
```
Was returned as `'\n        just a comment containing @ts-ignore somewhere in a block\n      '`